core/sql/parser/sqlparser.y

```diff
@@ -6647,6 +6652,26 @@ table_as_tmudf_function : TOK_UDF '(' table_mapping_function_invocation ')'
 
 table_name_and_hint : table_name optimizer_hint hbase_access_options
                       {
+                        NAString tmp = ((*$1).getQualifiedNameAsString());
+                        if(SqlParser_CurrentParser->with_clauses_->contains(&tmp) )
+                        {
+                          RelExpr *re = SqlParser_CurrentParser->with_clauses_->getFirstValue(&tmp);
+                          $$=re->copyTree(PARSERHEAP());
```

Comment:
yes, I am not quite sure how memory was managed in the parser, I  changed to delete $1